src/main/java/org/folio/dataexp/service/DownloadRecordService.java

```diff
@@ -0,0 +1,71 @@
+package org.folio.dataexp.service;
+
+
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dataexp.exception.export.DownloadRecordException;
+import org.folio.dataexp.service.export.ExportStrategyFactory;
+import org.folio.dataexp.service.export.S3ExportsUploader;
+import org.folio.dataexp.service.export.strategies.JsonToMarcConverter;
+import org.folio.spring.FolioExecutionContext;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+@Log4j2
+public class DownloadRecordService {
+
+  private final ExportStrategyFactory exportStrategyFactory;
+  private final JsonToMarcConverter jsonToMarcConverter;
+  private final S3ExportsUploader s3Uploader;
+  private final InputFileProcessor inputFileProcessor;
+  protected final FolioExecutionContext folioExecutionContext;
+
+  public ByteArrayResource processRecordDownload(final UUID recordId,
```

Comment:
Can InputStreamResource be returned here instead of ByteArrayResource? Possibly, this will allow to operate with InputStream. See comment for method **public String readMarcFile(String dirName)**.

Reply:
done